### PR TITLE
[ECF-1-264] ECT mentor estimates

### PR DESCRIPTION
**Pull request by amazimbe**
### Context
As a newly registered induction coordinator
Given that I have selected the full induction programme for my school
I should be able to enter estimates for the number of mentors and early career teachers in the next cohort

### Changes proposed in this pull request
Refactoring choices controller. I have added an induction_programme namespace and refactored accordingly.
Added estimates controller

### CODE REVIEW

File: app/controllers/induction_programme/estimates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      SchoolCohort.find_or_create_by!(cohort: @cohort, school: @school)
+    end
+  end
+
+  def cohort_estimates
+    params.require(:school_cohort_form)
+          .permit(
+            :estimated_teacher_count,
+            :estimated_mentor_count,
+          )
```
Comment (kicferk1) [inline app/controllers/induction_programme/estimates_controller.rb:35]: If we do multiline, can we have `require` on a new line too?

File: app/controllers/induction_programme/estimates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+private
+
+  def school_cohort
+    @school_cohort ||= begin
+      # TODO: Register and Partner 262: Figure out how to update current year
+      @cohort = Cohort.find_by(start_year: 2021)
+      @school = current_user.induction_coordinator_profile.schools.first
+      SchoolCohort.find_or_create_by!(cohort: @cohort, school: @school)
```
Comment (kicferk1) [inline app/controllers/induction_programme/estimates_controller.rb:26]: Do these need to be `@` variables?

File: app/forms/school_cohort_form.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class SchoolCohortForm
+  include ActiveModel::Model
+
+  attr_accessor :estimated_mentor_count, :estimated_teacher_count
+
+  validates :estimated_mentor_count, presence: { message: "Enter your expected number of mentors" }
+  validates :estimated_teacher_count, presence: { message: "Enter your expected number of ECTs" }
```
Comment (kicferk1) [inline app/forms/school_cohort_form.rb:9]: Do we want 'more or equal than 0' validation here?

File: app/views/induction_programme/estimates/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          :estimated_mentor_count,
+          width: 20,
+          link_errors: true,
+          label: {
+            text: "Number of mentors",
+            class: "govuk-label govuk-label--m"
+          }
+      ) %>
+
+
```
Comment (kicferk1) [inline app/views/induction_programme/estimates/show.html.erb:27]: I'm a whitespace hater

File: app/forms/school_cohort_form.rb (hunk 3 above)
Comment (amazimbe) [reply app/forms/school_cohort_form.rb:9]: sure

File: app/controllers/induction_programme/estimates_controller.rb (hunk 2 above)
Comment (amazimbe) [reply app/controllers/induction_programme/estimates_controller.rb:26]: Which one SchoolCohort

Comment (kicferk1) [reply app/controllers/induction_programme/estimates_controller.rb:26]: I meant `@school` and `@cohort` - I don't think they are used outside of this method (or am I wrong?)

File: spec/forms/school_cohort_form_spec.rb [hunk 5]
```diff
@@ -6,5 +6,13 @@
...
     it { is_expected.to validate_presence_of(:estimated_mentor_count).with_message("Enter your expected number of mentors") }
     it { is_expected.to validate_presence_of(:estimated_teacher_count).with_message("Enter your expected number of ECTs") }
+
+    it do
+      should validate_numericality_of(:estimated_mentor_count).is_greater_than_or_equal_to(0)
+    end
+
+    it do
+      should validate_numericality_of(:estimated_teacher_count).is_greater_than_or_equal_to(0)
+    end
```
Comment (kicferk1) [inline spec/forms/school_cohort_form_spec.rb]: Can we keep the style of the tests the same?
```
    it { is_expected.to validate_numericality_of(:estimated_teacher_count).is_greater_than_or_equal_to(0) }
```

File: app/forms/school_cohort_form.rb [hunk 6]
```diff
@@ -7,4 +7,6 @@ class SchoolCohortForm
 
   validates :estimated_mentor_count, presence: { message: "Enter your expected number of mentors" }
   validates :estimated_teacher_count, presence: { message: "Enter your expected number of ECTs" }
+  validates :estimated_mentor_count, numericality: { greater_than_or_equal_to: 0 }
+  validates :estimated_teacher_count, numericality: { greater_than_or_equal_to: 0 }
```
Comment (kicferk1) [inline app/forms/school_cohort_form.rb]: We could do it in the same lines I think.
```
validates :estimated_mentor_count, presence: { message: "Enter your expected number of mentors" }, numericality: { greater_than_or_equal_to: 0 }
```

Unless you think it makes things less readable of course.

Comment (amazimbe) [reply app/forms/school_cohort_form.rb]: I think it's better separated. Shorter, so I don't need to worry about splitting to multiple lines and easier to read too

**Review by kicferk1 — APPROVED** [review]

**Review by rupollbabu — APPROVED** [review]
Currently It is possible to create/register only one induction coordinator per school(with URN).Test covers for estimates count number of ECTs and number of mentors.

**Pull request merged**